torch/autograd/_functions/pointwise.py

```diff
@@ -138,103 +156,123 @@ def backward(ctx, grad_output):
+        ctx.save_for_backward(i)
         return i.sqrt()
 
-    def backward(self, grad_output):
-        i, = self.saved_tensors
+    @staticmethod
+    def backward(ctx, grad_output):
+        i, = ctx.saved_variables
         return grad_output.mul(i.pow(-0.5)).div(2)
 
 
 class Sin(Function):
 
-    def forward(self, i):
-        self.save_for_backward(i)
+    @staticmethod
+    def forward(ctx, i):
+        ctx.save_for_backward(i)
         return i.sin()
 
-    def backward(self, grad_output):
-        i, = self.saved_tensors
+    @staticmethod
+    def backward(ctx, grad_output):
+        i, = ctx.saved_variables
         return grad_output * i.cos()
 
 
 class Cos(Function):
 
-    def forward(self, i):
-        self.save_for_backward(i)
+    @staticmethod
+    def forward(ctx, i):
+        ctx.save_for_backward(i)
         return i.cos()
 
-    def backward(self, grad_output):
-        i, = self.saved_tensors
+    @staticmethod
+    def backward(ctx, grad_output):
+        i, = ctx.saved_variables
         return grad_output.mul(i.sin()).neg_()
 
 
 class Tan(Function):
 
-    def forward(self, i):
-        self.save_for_backward(i)
+    @staticmethod
+    def forward(ctx, i):
+        ctx.save_for_backward(i)
         return i.tan()
 
-    def backward(self, grad_output):
-        i, = self.saved_tensors
+    @staticmethod
+    def backward(ctx, grad_output):
+        i, = ctx.saved_variables
         return grad_output.div(i.cos().pow(2))
 
 
 class Asin(Function):
 
-    def forward(self, i):
-        self.save_for_backward(i)
+    @staticmethod
+    def forward(ctx, i):
+        ctx.save_for_backward(i)
         return i.asin()
 
-    def backward(self, grad_output):
-        i, = self.saved_tensors
-        return grad_output * (1 - i.mul(i)).sqrt_().reciprocal_()
+    @staticmethod
+    def backward(ctx, grad_output):
+        i, = ctx.saved_variables
+        return grad_output * (1 - i.mul(i)).sqrt().reciprocal()
 
 
 class Acos(Function):
 
-    def forward(self, i):
-        self.save_for_backward(i)
+    @staticmethod
+    def forward(ctx, i):
+        ctx.save_for_backward(i)
         return i.acos()
 
-    def backward(self, grad_output):
-        i, = self.saved_tensors
-        return grad_output.mul((1 - i.mul(i)).sqrt_().reciprocal_()).neg_()
+    @staticmethod
+    def backward(ctx, grad_output):
+        i, = ctx.saved_variables
+        return grad_output.mul((1 - i.mul(i)).sqrt().reciprocal()).neg()
 
 
 class Atan(Function):
 
-    def forward(self, i):
-        self.save_for_backward(i)
+    @staticmethod
+    def forward(ctx, i):
+        ctx.save_for_backward(i)
         return i.atan()
 
-    def backward(self, grad_output):
-        i, = self.saved_tensors
-        return grad_output * i.mul(i).add_(1).reciprocal_()
+    @staticmethod
+    def backward(ctx, grad_output):
+        i, = ctx.saved_variables
+        return grad_output * i.mul(i).add(1).reciprocal()
 
 
 class Reciprocal(Function):
```

Comment:
Can you add a TODO saying "make inplace and update grad formulas" here? We could add an inplace version and make many of the definitons you modified more efficient by using `reciprocal_()`